Library/Homebrew/compilers.rb

```diff
@@ -6,6 +6,7 @@ module CompilerConstants
     "gcc-4.0"  => :gcc_4_0,
     "gcc-4.2"  => :gcc,
     "clang"    => :clang,
+    "brew_clang" => :brew_clang,
```

Comment:
How about `:clang_3_8` since
```
lrwxr-xr-x  1 joe  admin         9 Jun  3 20:43 clang -> clang-3.8
lrwxr-xr-x  1 joe  admin         5 Jun  3 20:43 clang++ -> clang
-r-xr-xr-x  1 joe  admin  44760792 Jun  3 20:43 clang-3.8
```